### PR TITLE
Graph node micro-optimisation

### PR DESCRIPTION
**Pull request by e-strokov**
Node _dirtify method was deoptimised by compiler due to num. agrs. ambiguity.
Same for Mat4 and Mat3 constructors. Requires user-code changes.  

I confirm I have signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

### CODE REVIEW

**Comment by willeastcott** [comment]
Can you post some performance numbers that demonstrate that this change is at least as fast as the current implementation?

File: src/math/mat3.js [hunk 1]
```diff
@@ -78,6 +60,33 @@ Object.assign(pc, (function () {
...
         },
 
+        /**
+         * @function
+         * @name pc.Mat3#set
+         * @description Copies the contents of a source array[9] to a destination 3x3 matrix.
+         * @param {Array} src An array[9] to be copied.
+         * @returns {pc.Mat3} Self for chaining
+         * @example
+         * var src = [0, 1, 2, 3, 4, 5, ,6, 7, 8];
```
Comment (willeastcott) [inline src/math/mat3.js]: Extra comma needs to be removed.

File: src/math/mat3.js [hunk 2]
```diff
@@ -78,6 +60,33 @@ Object.assign(pc, (function () {
             return this;
         },
 
+        /**
+         * @function
+         * @name pc.Mat3#set
+         * @description Copies the contents of a source array[9] to a destination 3x3 matrix.
+         * @param {Array} src An array[9] to be copied.
```
Comment (willeastcott) [inline src/math/mat3.js]: I think my preference here is to have the function take 9 number arguments than 1 array. The problem being that we already have pc.Mat4#set that takes an array. As you can see below where you have updated the creation of pc.Mat3#ZERO, you've had to allocate an array redundantly. I wonder how many people currently use pc.Mat4#set...

**Comment by willeastcott** [comment]
On reflection, maybe it would have been better to create 2 PRs: Matrix optimization and syncHierarchy optimization. But I probably wouldn't bother splitting it now that it's been created. 😄

File: src/math/mat3.js (hunk 2 above)
Comment (e-strokov) [reply src/math/mat3.js]: It's not a problem at all to change this - my intention was to keep M3 and M4 interface consistent. So let me know if 9 args is more preferable over an array.

Comment (willeastcott) [reply src/math/mat3.js]: As discussed, let's leave this as a single array argument for now, then.

**Pull request merged**